api/src/main/java/org/openmrs/BaseOpenmrsObject.java

```diff
@@ -79,6 +82,11 @@ public boolean equals(Object obj) {
 		if (!(obj instanceof BaseOpenmrsObject)) {
 			return false;
 		}
+		//In case of hibernate proxy objects we need to get real classes
+		Class<?> thisClass = Hibernate.getClass(this);

```

Comment:
Just to check, are we sure that this will work if it is *not* a Hibernate-managed class?